D1 and D2 fail to build on Linux after 86adfc0

While D1 and D2 build successfully in Windows using Visual Studio 2022 (up to the latest commit - d891e79), changes introduced in 86adfc0 break the build on Linux. I'm using GCC 13.3.0 on Ubuntu 24.04.1 LTS (Noble Numbat) and reverting 86adfc0 fixes the build issues. The core of the problem seems to be the use of `sprintf_s` and `strcpy_s` in [`newdemo.c`](https://github.com/dxx-redux/dxx-redux/blob/main/d1/main/newdemo.c)

```
/home/adolfo/CLionProjects/dxx-redux/d1/main/newdemo.c:3263:(.text+0xfa9): undefined reference to `sprintf_s'
/usr/bin/ld: /home/adolfo/CLionProjects/dxx-redux/d1/main/newdemo.c:3269:(.text+0x100c): undefined reference to `sprintf_s'
/usr/bin/ld: /home/adolfo/CLionProjects/dxx-redux/d1/main/newdemo.c:3283:(.text+0x1058): undefined reference to `sprintf_s'
/usr/bin/ld: /home/adolfo/CLionProjects/dxx-redux/d1/main/newdemo.c:3281:(.text+0x108a): undefined reference to `sprintf_s'
/usr/bin/ld: /home/adolfo/CLionProjects/dxx-redux/d1/main/newdemo.c:3287:(.text+0x10b5): undefined reference to `sprintf_s'
/usr/bin/ld: CMakeFiles/d1x-redux.dir/newdemo.c.o:/home/adolfo/CLionProjects/dxx-redux/d1/main/newdemo.c:3296: more undefined references to `sprintf_s' follow
/usr/bin/ld: CMakeFiles/d1x-redux.dir/newdemo.c.o: in function `newdemo_get_default_filename':
/home/adolfo/CLionProjects/dxx-redux/d1/main/newdemo.c:3289:(.text+0x112c): undefined reference to `strcpy_s'
/usr/bin/ld: /home/adolfo/CLionProjects/dxx-redux/d1/main/newdemo.c:3252:(.text+0x1162): undefined reference to `sprintf_s'
/usr/bin/ld: /home/adolfo/CLionProjects/dxx-redux/d1/main/newdemo.c:3254:(.text+0x1184): undefined reference to `sprintf_s'
/usr/bin/ld: CMakeFiles/d1x-redux.dir/newdemo.c.o: in function `newdemo_prompt_filename':
/home/adolfo/CLionProjects/dxx-redux/d1/main/newdemo.c:3304:(.text+0x1248): undefined reference to `strncpy_s'
/usr/bin/ld: /home/adolfo/CLionProjects/dxx-redux/d1/main/newdemo.c:3334:(.text+0x137c): undefined reference to `strcpy_s'
/usr/bin/ld: CMakeFiles/d1x-redux.dir/newdemo.c.o: in function `newdemo_stop_recording':
/home/adolfo/CLionProjects/dxx-redux/d1/main/newdemo.c:3364:(.text+0x146e): undefined reference to `sprintf_s'
collect2: error: ld returned 1 exit status
gmake[2]: *** [main/CMakeFiles/d1x-redux.dir/build.make:1197: main/d1x-redux] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:410: main/CMakeFiles/d1x-redux.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
```

--- a/d1/include/strutil.h
+++ b/d1/include/strutil.h
@@ -34,4 +34,9 @@ int string_array_sort_func(char **e0, char **e1);
 // reallocate pointers to save memory, sort list alphabetically and remove duplicates according to 'comp'
 void string_array_tidy(char ***list, char **list_buf, int *num_str, int *max_str, int *max_buf, int offset, int (*comp)( const char *, const char * ));
 
+#ifndef WIN32
+int strcpy_s(char *dest, size_t dest_size, const char *src);
+int sprintf_s(char *buffer, size_t buffer_size, const char *format, ...);
+#endif
+
 #endif /* _STRUTILS_H */

--- a/d1/main/newdemo.c
+++ b/d1/main/newdemo.c
@@ -3301,7 +3301,7 @@ void newdemo_get_default_filename(char* filename_buffer, unsigned int filename_b
 int newdemo_prompt_filename(char* filename_buffer, unsigned int filename_buffer_length)
 {
 	char filename_from_prompt[PATH_MAX] = "";
-	strncpy_s(filename_from_prompt, SDL_arraysize(filename_from_prompt), filename_buffer, filename_buffer_length);
+	strcpy_s(filename_from_prompt, SDL_arraysize(filename_from_prompt), filename_buffer);
 
 	do {
 		newmenu_item m[2];

--- a/d1/misc/strutil.c
+++ b/d1/misc/strutil.c
@@ -323,3 +323,26 @@ void string_array_tidy(char ***list, char **list_buf, int *num_str, int *max_str
 
 	*num_str = j;
 }
+
+#ifndef WIN32
+int strcpy_s(char *dest, size_t dest_size, const char *src)
+{
+	if (!memchr(src, 0, dest_size)) // src longer than dest_size?
+		abort();
+	strcpy(dest, src);
+	return 0;
+}
+
+int sprintf_s(char *buffer, size_t buffer_size, const char *format, ...)
+{
+	va_list vp;
+	int ret;
+
+	va_start(vp, format);
+	ret = vsnprintf(buffer, buffer_size, format, vp);
+	va_end(vp);
+	if (ret >= buffer_size) // result didn't fit?
+		abort();
+	return ret;
+}
+#endif

--- a/d2/include/strutil.h
+++ b/d2/include/strutil.h
@@ -32,4 +32,9 @@ int string_array_sort_func(char **e0, char **e1);
 // reallocate pointers to save memory, sort list alphabetically and remove duplicates according to 'comp'
 void string_array_tidy(char ***list, char **list_buf, int *num_str, int *max_str, int *max_buf, int offset, int (*comp)( const char *, const char * ));
 
+#ifndef WIN32
+int strcpy_s(char *dest, size_t dest_size, const char *src);
+int sprintf_s(char *buffer, size_t buffer_size, const char *format, ...);
+#endif
+
 #endif /* _STRUTILS_H */

--- a/d2/main/newdemo.c
+++ b/d2/main/newdemo.c
@@ -3535,7 +3535,7 @@ void newdemo_get_default_filename(char* filename_buffer, unsigned int filename_b
 int newdemo_prompt_filename(char* filename_buffer, unsigned int filename_buffer_length)
 {
 	char filename_from_prompt[PATH_MAX] = "";
-	strncpy_s(filename_from_prompt, SDL_arraysize(filename_from_prompt), filename_buffer, filename_buffer_length);
+	strcpy_s(filename_from_prompt, SDL_arraysize(filename_from_prompt), filename_buffer);
 
 	do {
 		newmenu_item m[2];

--- a/d2/misc/strutil.c
+++ b/d2/misc/strutil.c
@@ -311,3 +311,26 @@ void string_array_tidy(char ***list, char **list_buf, int *num_str, int *max_str
 
 	*num_str = j;
 }
+
+#ifndef WIN32
+int strcpy_s(char *dest, size_t dest_size, const char *src)
+{
+	if (!memchr(src, 0, dest_size)) // src longer than dest_size?
+		abort();
+	strcpy(dest, src);
+	return 0;
+}
+
+int sprintf_s(char *buffer, size_t buffer_size, const char *format, ...)
+{
+	va_list vp;
+	int ret;
+
+	va_start(vp, format);
+	ret = vsnprintf(buffer, buffer_size, format, vp);
+	va_end(vp);
+	if (ret >= buffer_size) // result didn't fit?
+		abort();
+	return ret;
+}
+#endif

